contracts/tasks/helpers/Prover.ts

```diff
@@ -80,37 +73,42 @@ export class Prover {
    */
   async proveMessageProcessing(proofs: Proof[]): Promise<void> {
     // retrieve the values we need from the smart contracts
-    const [treeDepths, numSignUpsAndMessages, numBatchesProcessed, stateTreeDepth, dd, coordinatorPubKeyHash, mode] =
-      await Promise.all([
-        this.pollContract.treeDepths(),
-        this.pollContract.numSignUpsAndMessages(),
-        this.mpContract.numBatchesProcessed().then(Number),
-        this.maciContract.stateTreeDepth().then(Number),
-        this.pollContract.getDeployTimeAndDuration(),
-        this.pollContract.coordinatorPubKeyHash(),
-        this.mpContract.mode(),
-      ]);
+    const [
+      treeDepths,
+      batchSizes,
+      numSignUpsAndMessages,
+      numBatchesProcessed,
+      stateTreeDepth,
+      dd,
+      coordinatorPubKeyHash,
+      mode,
+    ] = await Promise.all([
+      this.pollContract.treeDepths(),
+      this.pollContract.batchSizes(),
+      this.pollContract.numSignUpsAndMessages(),
+      this.mpContract.numBatchesProcessed().then(Number),
+      this.maciContract.stateTreeDepth().then(Number),
+      this.pollContract.getDeployTimeAndDuration(),
+      this.pollContract.coordinatorPubKeyHash(),
+      this.mpContract.mode(),
+    ]);
 
     const numSignUps = Number(numSignUpsAndMessages[0]);
     const numMessages = Number(numSignUpsAndMessages[1]);
-    const messageBatchSize = STATE_TREE_ARITY ** Number(treeDepths[1]);
+    const messageBatchSize = MESSAGE_BATCH_SIZE;
```

Comment:
this should not be a constant it should be taken from the poll contract

Reply:
resolved